esmvalcore/cmor/_fixes/icon/icon.py

```diff
@@ -0,0 +1,163 @@
+
+    @staticmethod
+    def _fix_height(cube, plev_points_cube, plev_bounds_cube):
+        """Fix height coordinate of cube."""
+        air_pressure_points = plev_points_cube.core_data()
+
+        # Get bounds from half levels and reshape array
+        air_pressure_bounds = plev_bounds_cube.core_data()
+        air_pressure_bounds = da.stack(
+            (air_pressure_bounds[:, :-1], air_pressure_bounds[:, 1:]), axis=-1)
+
+        # Setup air pressure coordinate with correct metadata and add to cube
+        air_pressure_coord = iris.coords.AuxCoord(
+            air_pressure_points,
+            bounds=air_pressure_bounds,
+            var_name='plev',
+            standard_name='air_pressure',
+            long_name='pressure',
+            units=plev_points_cube.units,
+            attributes={'positive': 'down'},
+        )
+        cube.add_aux_coord(air_pressure_coord, np.arange(cube.ndim))
+
+        # Add 'positive' attribute to height so iris can determine Z axis
+        # correctly
+        cube.coord('height').attributes['positive'] = 'down'
+
+        # Reverse height axis so that index 0 is surface level
+        cube = iris.util.reverse(cube, 'height')
+        return cube
+
+    @staticmethod
+    def _fix_lat_lon(cube, lat_name, lon_name):
+        """Fix latitude and longitude coordinates of cube."""
+        lat = cube.coord(lat_name)
+        lon = cube.coord(lon_name)
+
+        # Fix metadata
+        lat.var_name = "lat"
+        lon.var_name = "lon"
+        lat.standard_name = "latitude"
+        lon.standard_name = "longitude"
+        lat.long_name = "latitude"
+        lon.long_name = "longitude"
+        lat.convert_units('degrees')
+        lon.convert_units('degrees')
+        lat.units = "degrees_north"
+        lon.units = "degrees_east"
+
+        # If grid is not unstructured, no further changes are necessary
+        if cube.coord_dims(lat) != cube.coord_dims(lon):
+            return
+        horizontal_coord_dims = cube.coord_dims(lat)
+        if len(horizontal_coord_dims) != 1:
+            return
+
+        # Add dimension name for cell index used to store the unstructured grid
+        index_coord = iris.coords.DimCoord(
+            np.arange(cube.shape[horizontal_coord_dims[0]]),
+            var_name="i",
+            long_name=("first spatial index for variables stored on an "
+                       "unstructured grid"),
+            units="1",
+        )
+        cube.add_dim_coord(index_coord, horizontal_coord_dims)
+
+    @staticmethod
+    def _fix_time(cube):
+        """Fix time coordinate of cube."""
+        t_coord = cube.coord("time")
+        t_coord.var_name = 'time'
+        t_coord.standard_name = 'time'
+        t_coord.long_name = 'time'
+
+        # Convert units to CF format
+        t_unit = t_coord.attributes.pop("invalid_units")
+        timestep, _, t_fmt_str = t_unit.split(" ")
+        new_t_unit_str = f"{timestep} since 1850-01-01"
```

Comment:
In the ICON files that I have `{timestep}` is `day`, but I think we can leave it like this to support different units. Is there any reason to not use 1850-01-01 as the baseline?

Reply:
Hm. Looking at the example I think I might be even stricter here and match this against the fixed string `day as %Y%m%d.%f`. I mean, what could you envision as an alternative? `hour as ???`

Also, looking a bit closer at the documentation of [`datetime.strptime`](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes), I worry a wee bit about the `%f`. Do you perhaps have some sub-daily data (6hr, 3hr, ...) to check that everything works out there?

Reply:
On the basetime, well, CMIP6 makes a bunch of provisions (see the note linked above), but it's not critical, I think.

Reply:
Unfortunately, I don't have sub-daily data to test this right now. I will fix the invalid units and add a check. If it doesn't work out of the box, support for sub-daily data can be implemented at a later stage

Reply:
```suggestion
        new_t_unit_str = f"days since 1850-01-01"
```
 CMIP6 requires the plural form "days", which also makes more sense to me. If you want to keep it tied to the `timestep`, I would introduce a translation dictionary like `dict(day="days")`, but honestly, I think keeping it to literally "days" here is the simplest and only if and when we have an example that doesn't come as "day" we'll know how to do better.